### PR TITLE
cs/refactor/bulk dupliates have fullname

### PR DESCRIPTION
**Pull request by chancesmith**
## Purpose of PR

Some contacts share the same first initial and last name.

## Changes Made

- use full name as file name if initial already exists
- error if initial and fullname is already used
- show a warning or a error for each case above

## Screenshots

error
Screen Shot 2023-01-06 at 1 07 29 PM

warnings
Screen Shot 2023-01-06 at 12 50 35 PM

### CODE REVIEW

File: csv-bulk-signature/src/main.ts [hunk 1]
```diff
@@ -90,21 +90,53 @@ async function createSignatureFile(
...
+  }
+
+  if (!fs.existsSync(fullNameFilePath)) {
+    return createFileWithFullName(fullNameFilePath, fullName, fileName, html);
   }
+
+  console.error(
+    `  🔴 ERROR: For ${fullName}, ${fileName}.htm and ${fullNameFileName}.htm already exist.`
+  );
+}
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:108]: - refactor to handle one case where fullname is a duplicate too
- show a warning for the first duplicate
- show an error if both options are taken

File: csv-bulk-signature/src/main.ts [hunk 2]
```diff
@@ -90,21 +90,53 @@ async function createSignatureFile(
...
+
+async function createFile(filePath: string, html: string) {
+  await fs.promises.writeFile(filePath, html);
+}
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:112]: refactor for file creation

File: csv-bulk-signature/src/main.ts [hunk 3]
```diff
@@ -90,21 +90,53 @@ async function createSignatureFile(
...
+  fullNameFilePath: string,
+  fullName: string,
+  fileName: string,
+  html: string
+) {
+  await createFile(fullNameFilePath, html);
+  console.error(
+    `  🟡 WARNING: For ${fullName}, ${fileName}.htm already exists. Instead ${fileName}.htm was created.`
+  );
+}
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:124]: create a file with the full name

File: csv-bulk-signature/src/main.ts [hunk 4]
```diff
@@ -90,21 +90,53 @@ async function createSignatureFile(
...
+
+function getFullNameFileName(contact: Contact) {
+  const fullName = contact["Full Name*"];
+  const lowercaseFullName = fullName.toLowerCase();
+  const fullNameFileName = lowercaseFullName.replace(/\s/g, "_");
+  return { fullNameFileName, fullName };
 }
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:131]: building up full name file name

File: csv-bulk-signature/src/main.ts [hunk 5]
```diff
@@ -90,21 +90,53 @@ async function createSignatureFile(
...
 
 function getFileName(contact: Contact) {
   const nameSplit = contact["Full Name*"].split(" ");
   const firstInitial = nameSplit[0].charAt(0);
   const lastName = nameSplit[1];
   const fileName = `${firstInitial}${lastName}`;
-  return fileName;
+  const lowerCaseFullName = fileName.toLowerCase();
+  return lowerCaseFullName;
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:139]: lowercase filename to match what we had in v1

**Review by adamnitti — APPROVED** [review]

**Pull request merged**